src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -420,24 +421,25 @@ public TypeWithAnnotations InferReturnType(ConversionsBase conversions, NamedTyp
 
     internal abstract class UnboundLambdaState
     {
-        private UnboundLambda _unboundLambda; // we would prefer this readonly, but we have an initialization cycle.
+        private UnboundLambda _unboundLambda = null!; // we would prefer this readonly, but we have an initialization cycle.
```

Comment:
>_unboundLambda [](start = 30, length = 14)

It feels like this should be `?`, and the property should assert that `_unboundLambda` is not null on return.

Reply:
`_unboundLambda` is used in many locations that all expect the value to be not `null`. This seemed like a reasonable shortcut.